state_machine/applications/system-check/tests/imu_test.py

```diff
@@ -6,215 +6,276 @@
 """
 
 import time
-from ulab.numpy.linalg import norm
-import random
+from ulab import numpy
 from lib import pycubed as cubesat
 
 
 wait_time = 3
+norm = numpy.linalg.norm
 
 
-def still_user_test():
+def request_imu_data(prompt):
     """
-    All user interaction for the still test happens in this function
-    Set wait times, prompt the users and gather IMU data and input
+    Ask the user to type y/any key to start/cancel the test
+    If y, gather IMU data
+    Accelerometer and Gyroscope readings are collected 10 times
+    in wait_time seconds
+    Magnetometer readings are collected at the start and end
+    of wait_time seconds
+    If any other key, return None for all readings
     """
-    # print user prompts
-    print("Please leave the cubesat flat on a table." +
-          "Waiting {wait_time} seconds.")
-    time.sleep(wait_time)
+    # ask user whether or not to start the test
+    print(prompt, end="")
+    start_test = input("Type Y to start the test, any key to cancel: ")
+
+    # if user opts to cancel the test, return None, handle in the next function
+    if start_test.lower() != "y":
+        return None, None, None
+
+    # else proceed with the test
+    time.sleep(0.5)
     print("Collecting IMU data...")
 
-    # collect acceleration and gyroscope readings, return 0.0 mag default
-    time.sleep(wait_time)
-    acc = cubesat.acceleration()
-    gyro = cubesat.gyro()
+    # collect magnetometer reading at the start of wait time
+    start_mag = cubesat.magnetic()
+
+    # collect average accelerometer and gyroscope readings
+    # do 10 reads over wait_time
+    acc_total = [0, 0, 0]
+    gyro_total = [0, 0, 0]
+    for i in range(10):
+        acc = cubesat.acceleration()
+        gyro = cubesat.gyro()
+        acc_total[0] += acc[0]
+        acc_total[1] += acc[1]
+        acc_total[2] += acc[2]
+        gyro_total[0] += gyro[0]
+        gyro_total[1] += gyro[1]
+        gyro_total[2] += gyro[2]
+        time.sleep(wait_time / 10)
+
+    # collect magnetometer reading at the end of wait time
+    end_mag = cubesat.magnetic()
+
+    # calculate the average acc and gyro readings
+    acc_final = tuple(val / 10 for val in acc_total)
+    gyro_final = tuple(val / 10 for val in gyro_total)
+
+    # calculate the change in the total magnetometer reading
+    sub_mag = tuple(map(lambda end, start: end - start, end_mag, start_mag))
```

Comment:
ulab/numpy arrays directly support element-wise subtraction (like vectors in math/physics).

```
start_mag = numpy.array(cubesat.magnetic())
end_mag = numpy.array(cubesat.magnetic())
mag = norm(end_mag - start_mag)
```

Reply:
Thanks!